modules/core/src/test/java/org/apache/ignite/internal/processors/cache/SafeLogTxFinishErrorTest.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.distributed.near.GridNearTxLocal;
+import org.apache.ignite.internal.processors.cache.transactions.IgniteInternalTx;
+import org.apache.ignite.internal.processors.cache.transactions.IgniteTxAdapter;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.transactions.Transaction;
+import org.junit.Test;
+
+import static java.lang.String.format;
+import static java.util.Objects.nonNull;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.internal.util.IgniteUtils.compact;
+import static org.apache.ignite.testframework.GridTestUtils.getFieldValue;
+import static org.apache.ignite.testframework.GridTestUtils.setFieldValue;
+
+/**
+ * Tests verifying that {@link IgniteTxAdapter#logTxFinishErrorSafe}
+ * runs without errors.
+ */
+public class SafeLogTxFinishErrorTest extends GridCommonAbstractTest {
+    /** Logger for listen log messages. */
+    private final ListeningTestLogger log = new ListeningTestLogger(false, super.log);
+
+    /** Flag to remove the FailureHandler when creating configuration for node. */
+    private boolean removeFailureHandler;
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setCacheConfiguration(
+            new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+                .setAffinity(new RendezvousAffinityFunction(false, 32))
+                .setAtomicityMode(TRANSACTIONAL)
+        );
+
+        if (removeFailureHandler)
+            cfg.setFailureHandler(null);
+
+        return cfg;
+    }
+
+    /**
+     * Checking that {@link IgniteTxAdapter#logTxFinishErrorSafe} is performed
+     * without errors with FailureHandler.
+     *
+     * @throws Exception If any error occurs.
+     */
+    @Test
+    public void testSafeLogTxFinishErrorWithFailureHandler() throws Exception {
+        checkSafeLogTxFinishError();
+    }
+
+    /**
+     * Checking that {@link IgniteTxAdapter#logTxFinishErrorSafe} is performed
+     * without errors without FailureHandler.
+     *
+     * @throws Exception If any error occurs.
+     */
+    @Test
+    public void testSafeLogTxFinishErrorWithoutFailureHandler()throws Exception {
+        removeFailureHandler = true;
+
+        checkSafeLogTxFinishError();
+    }
+
```

Comment:
Seems that we should add more tests in order to check catch clause of logTxFinishErrorSafe()
```
catch (Throwable e0) {
            e.addSuppressed(e0);

            U.error(log, String.format(fmt, commit, CU.txString(this)), e);
        }
```